### ISSUE TITLE
Detect math constants

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1656]
# Rule request

## Thesis

The following constants must be imported from `math` module instead of being defined in the user code: pi, e, tau, inf, nan. For example:

```python
# bad:

PI = 3.1415

def area(r):
  return PI*r**2/2

# good:

import math

def area(r):
  return math.pi*r**2/2
```

## implementation

+ The constant should be detected by value with 2 signs after the dot precision rather than by name. For example, `3.14` most probably is π.
+ The rule should handle the constant in any form, including the scientific one. For example, `3.141593e+00`.

## Reasoning

1. Avoid making mistakes in the number
1. `math.e` makes it clear that this is a mathematical constant rather than some random magic number.

## Related links

+ Related to #1383. 
+ Probably, the implementation of the issue above (#1654) should be extended to handle more cases. In that case, let's also forbid `float('inf')` in favor of `math.inf` as it was done with nan.
+ https://docs.python.org/3/library/math.html#constants

**Comment by orsinium** [comment]
I'm wondering if we should forbid `math.tau` in favor of `2 * math.pi` :thinking:

**Comment by orsinium** [comment]
Also, probably we should forbid `from math import pi` in favor of `import math`.

**Comment by sobolevn** [comment]
We already have this rule: https://github.com/wemake-services/wemake-python-styleguide/blob/9879a18acafc691db215afb7c42dceb9a365a604/wemake_python_styleguide/violations/best_practices.py#L1819

But, let's open a new issue for `import math`! 👍
